agent/ecr/factory.go

```diff
@@ -65,14 +66,16 @@ func (factory *ecrFactory) GetClient(authData *apicontainer.ECRAuthData) (ECRCli
 }
 
 // getClientConfig returns the config for the ecr client based on authData
-func getClientConfig(httpClient *http.Client, authData *apicontainer.ECRAuthData) (*aws.Config, error) {
+func getClientConfig(httpClient *http.Client, authData *apicontainer.ECRAuthData, ipCompatibility *ipcompatibility.IPCompatibility) (*aws.Config, error) {
```

Comment:
Why do we want `ipCompatibility` to be optional?

Reply:
Was a miss during implementation, corrected in latest set of changes